### ISSUE TITLE
Crash on login [Android]

### ISSUE THREAD

**Issue by agnithegreat** [myflashlab/GameServices-ANE#5]
I have created the test project to check GameServices-ANE.

On init it traces:

> [trace] onInit result = GameServices.SUCCESS
> [trace] canDoSilentLogin =  false
> [trace] If silent try is "false", it means that you have to login the user yourself using the "GameServices.login();" method.
> [trace] But if it's "true", it means that the user had signed in before and he will be signed in again silently shortly. and you have to wait for the "AuthEvents.LOGIN" event.

Logcat throws in same time:

> E/GooglePlayServicesUtil(14231): The Google Play services resources were not found. Check your project configuration to ensure that the resources are included.
> E/GooglePlayServicesUtil(14231): The Google Play services resources were not found. Check your project configuration to ensure that the resources are included.

Then I close the "ANE DEMO VERSION" popup and call GameServices.google.auth.login() and It crashes right away.

I have followed instructions and have set the actual APP_ID in descriptor.xml, set up all dependencies, but that still makes no sense. Also I have added Drive API as it said there: https://github.com/myflashlab/GameServices-ANE/issues/2. Could I miss something?

**Comment by myflashlab** [comment]
Ok, when it says ```The Google Play services resources were not found``` it means that you are missing something in your build. Make sure you have the following setup correctly:

1. Are you sure your AIR manifest is setup correctly? check every line [here](https://github.com/myflashlab/GameServices-ANE#air-xml-manifest) and make sure you are not missing anything.
2. Are you sure you have implemented all the dependency ANEs mentioned [here](https://github.com/myflashlab/GameServices-ANE#requirements)? The GameServices can't run without them.

Moreover, we have an [ANE Inspector tool](http://www.myflashlabs.com/a-tool-to-make-ane-integration-into-your-air-apps-a-breeze/) which you may find very useful when working with our ANEs.

**Comment by agnithegreat** [comment]
I have checked the manifest once again, everything is correct as said in Readme. Sure, you can check it there: https://www.dropbox.com/s/uuxdjdu0wcd1spu/Lotto-android.xml?dl=0.
Also I tried an ANE Inspector as well, and it says all dependencies are in their places. I have totally puzzled with this issue and I have to solve it.

**Comment by myflashlab** [comment]
You said the app will crash as soon as you call ```GameServices.google.auth.login()```, will you post the logs for the crash?

**Comment by agnithegreat** [comment]
How could I filter trash information from the log in logcat? There is a lot of stuff like WifiStateMachine, AALService, AlarmManager and so on showing, so it's becoming difficult to get something helpful. What is the best practices to use logcat in that case?
Also I could provide the APK.

**Comment by agnithegreat** [comment]
http://pastebin.com/dwvLjqvi - full log from app start to the crash
http://pastebin.com/VDWBhXYx - log with ignored tags ("MtkAudioLoud", "MALI    ", "AudioTrack", "wifiJNI ", "PowerManagerService", "SignalClusterView", "WifiConfigStore", "StatusBar.NetworkController", "Launcher3", "SurfaceView", "WifiTrafficPoller", "WifiHW  ", "wpa_supplicant", "NativeCrypto", "AALService", "AlarmManager", "AudioFlinger_Threads", "GraphicBuffer", "AudioPolicyManager", "libc-netbsd", "[MALI][Gralloc]", "BufferQueueProducer", "SocketClient", "libPerfService")

**Comment by myflashlab** [comment]
Ok, Thanks.

1. I was not able to locate the Android crash log lines! Android crash logs start with "caused by...." but I couldn't locate any!

2. You are not running your .apk on a shared package, are you? you should use the Captive build to be able to run the ANE correctly.

Moreover, to be able to find the problem, I think maybe it would be a good idea to create a demo project with using the ANE alone in it without using other ANEs. So your project will be a small one. That way, it would be a lot easier to analyze the logs and find the problem. Create a demo project with the GameServices ANE in it alone and see how things go.

**Comment by agnithegreat** [comment]
Okay, first of all:
1. I have sent you all logs I have got with logcat, and what I'm thinking, it could be not a crash, but application just goes behind, and it shows the Android desktop. This mostly happens when application don't have enough memory, same behaviour.
2. I always make a captive runtime or debug build, so it can't be a reason.

But finally:
I already tried to make an empty demo project to check the GameServices-ANE, but first time it didn't help much. So yesterday I tried to create it once again, added 2 buttons, 1st did `GameServices.init()`, 2nd calls `GameServices.google.auth.login();`. I was surprised, that login has been successful.
After that, I got back to the project and noticed, that when I'm trying to login in 1-2 seconds after init, it pass well, but if I wait more than that, the application goes to background without any message, and debugger just disconnects from it.
I really don't know, how to debug this behaviour, but it is the good step forward.

**Comment by agnithegreat** [comment]
Finally, I have an absolutely empty project with only GPGS ANE and all it's dependencies. I don't think there is something I could miss. But there are 2 ways of how to test it:
1. Start, press OK, wait 10 seconds, press the scene at any place. (Causes crash)
2. Start, press OK and press the scene ASAP (Successful login)

I have tested it on Lenovo Tab A10-70, even test project from this repository fails.
But BlueStacks Emulator didn't crash even after 30 seconds after init.
I don't know, is it a device problem, or anything else, I will do some tests on another devices and give you more info later.

APK:
https://www.dropbox.com/s/2weg2gbg75ahw9r/Lotto_Android_Test.apk?dl=0

**Comment by myflashlab** [comment]
I have downloaded the .apk file. We'll do some tests and try to reproduce the problem. We'll get back to you after some tests...

**Comment by agnithegreat** [comment]
I have found out, that the problem is in AIR SDK 22 I used.
There is an information about AIR SDK 20+ memory problems in this thread: http://forum.starling-framework.org/topic/does-starling-20-reduce-lost-context-issues
I have tried latest AIR SDK 23, and it helped much.

P.S. application shutdown also had the same problem with several other ANEs, like Rewarded Video ads. Looks like a memory overhead. Now, I think, the issue could be closed, but your users should know that AIR SDKs 20-22 could have such an issue.
